Source/AudioAnalysisTools/Private/Analyzers/BeatDetection.cpp

```diff
@@ -30,10 +40,21 @@ void UBeatDetection::UpdateFFTSubbandsSize(int32 InFFTSubbandsSize)
 	FFTVariance.SetNum(FFTSubbandsSize);
 	FFTBeatValues.SetNum(FFTSubbandsSize);
 	EnergyHistory.SetNum(FFTSubbandsSize);
+
+	// We resized the external array, so we have to resize the new array
+	UpdateEnergyHistorySize(EnergyHistorySize);
 }
 
 void UBeatDetection::UpdateEnergyHistorySize(int32 InEnergyHistorySize)
 {
+	// We'll assume nothing, and make sure our user has made a reasonable request
+	if (InEnergyHistorySize <= 0)
+	{
+		// Tell the user that they've tried to use an incorrect value, and where they tried it
+		UE_LOG(LogAudioAnalysis, Log, TEXT("Beat Detection energy history size '%d' is invalid, using default of 41"), InEnergyHistorySize);
+		return;
```

Comment:
Here, it would probably be better not to say "the default value of 41 is used", but "the value of %d remains" or something like that, since it can already be changed to another value.